apps/frontend/src/store/evaluations.ts

```diff
@@ -0,0 +1,87 @@
+import Store from '@/store/store';
+import {IEvaluation} from '@heimdall/interfaces';
+import axios from 'axios';
+import {
+  Action,
+  getModule,
+  Module,
+  Mutation,
+  VuexModule
+} from 'vuex-module-decorators';
+
+@Module({
+  namespaced: true,
+  dynamic: true,
+  store: Store,
+  name: 'EvaluationModule'
+})
+export class Evaluation extends VuexModule {
+  activeEvaluation: IEvaluation = {
+    id: '',
+    filename: '',
+    userId: '',
+    createdAt: new Date(),
+    updatedAt: new Date(),
+    data: {},
+    evaluationTags: []
+  };
+  activeTag: any = {};
+
+  @Action
+  setActiveEvaluation(evaluation: any) {
+    this.context.commit('SET_ACTIVE_EVALUATION', evaluation);
+  }
+
+  @Action
+  async deleteEvaluation() {
+    return axios.delete(`/evaluations/${this.activeEvaluation.id}`);
+  }
+
+  @Action({rawError: true})
+  async addTagToActiveEvaluation() {
+    return axios.post(
+      `/evaluation-tags/${this.activeEvaluation.id}`,
+      this.activeTag
+    );
+  }
+
+  @Action
+  setActiveEvaluationFilename(filename: string) {
+    return this.context.commit('UPDATE_FILENAME', filename);
+  }
+
+  @Action({rawError: true})
+  async deleteTag(tag: any) {
+    await axios.delete(`/evaluation-tags/${tag.id}`);
+  }
+
+  @Action({rawError: true})
+  async updateTag(tag: any) {
+    return axios.put(`/evaluation-tags/${tag.id}`, tag);
+  }
+
+  @Action({rawError: true})
+  async updateEvaluation() {
+    return axios
+      .put(`/evaluations/${this.activeEvaluation.id}`, this.activeEvaluation)
+      .then((response) => {
+        this.context.commit('SET_ACTIVE_EVALUATION', response.data);
+      });
+  }
```

Comment:
Normally the Vuex store is used for storing items that need to have their state shared across multiple components. This module seems to really exist for the purpose of passing the file to the Edit page. It seems like there are a few alternative ways to implement this:

1. Could the full list of evaluations be synced to the edit component with an ID of the current one to edit? https://vuejs.org/v2/guide/components-custom-events.html#sync-Modifier
2. Instead of storing only the active evaluation in Vuex, why not just store an array of all evaluations in Vuex and then pass the ID of the one you want to edit over to the edit component?

I just don't think syncing only one evaluation to Vuex is the best way to approach this.

Reply:
Ended up syncing all evaluations to the store, and passing the ID.